internal/persistence/sql/relationtuples.go

```diff
@@ -200,6 +200,42 @@ func (p *Persister) DeleteRelationTuples(ctx context.Context, rs ...*relationtup
 	})
 }
 
+func (p *Persister) DeleteAllRelationTuples(ctx context.Context, query *relationtuple.RelationQuery) error {
+	return p.Transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		sqlRawQuery := "DELETE FROM keto_relation_tuples WHERE"
+		args := []interface{}{}
+		if query.Namespace != "" {
+			n, err := p.GetNamespaceByName(ctx, query.Namespace)
+			if err != nil {
+				return err
+			}
+			sqlRawQuery += " namespace_id = ?"
+			args = append(args, n.ID)
+		}
+		if query.Object != "" {
+			sqlRawQuery += " AND object = ?"
+			args = append(args, query.Object)
+		}
+		if query.Relation != "" {
+			sqlRawQuery += " AND relation = ?"
+			args = append(args, query.Relation)
+		}
```

Comment:
Please use the pop query builder for this instead of raw query. Just as we do here: https://github.com/ory/keto/blob/7ee65b5cd0827a3de87be3b6a7fa62d9038ea2df/internal/persistence/sql/relationtuples.go#L218-L236
and here https://github.com/ory/keto/blob/7ee65b5cd0827a3de87be3b6a7fa62d9038ea2df/internal/persistence/sql/relationtuples.go#L186-L196

As is, there is no SQL injection possible, but it is easy to add one later if you don't pay attention.

Reply:
Ah, my bad. I actually used pop query builder but for some reason, it didn't work. Now I realized why :zany_face:
I will push a new commit.